### PR TITLE
ffmpeg: enable vaapi

### PR DESCRIPTION
**Pull request by Andarwinux**
Fully statically linked libva

### CODE REVIEW

**Comment by Andarwinux** [comment]
add libva-utils, need to build it manually.

**Comment by Andarwinux** [comment]
vaapi-win32 is not useful in mpv for the moment, only vaapi-copy is available due to interop not implemented, filters are not available.
But for ffmpeg, this unlocks a cross-vendor encoder and hardware filters.

**Comment by shinchiro** [comment]
What is `libva-utils` used for?

File: packages/libva-utils.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ExternalProject_Add(libva-utils
+    DEPENDS libva
```
Comment (shinchiro) [inline packages/libva-utils.cmake]: put libva under depends similar to other package file

**Comment by Andarwinux** [comment]
> What is `libva-utils` used for?

The tool used to query VAAPI device information.

**Comment by Andarwinux** [comment]
Added mesa to get the latest vaon12 driver.

**Comment by Andarwinux** [comment]
The directx-header can also be used for future d3d12va, although mingw-w64 already contains the relevant header, but it may not be updated in time.

**Comment by Andarwinux** [comment]
Fixed static linking problem.

**Pull request closed without merging**